### ISSUE TITLE
Unnecessary redirects

### ISSUE THREAD

**Issue by oliverjam** [fac19/week5-FKAM#45]
https://github.com/fac19/week5-FKAM/blob/5f48fe017de7a7bcf805c5b947330e8fdd3c1110/src/handlers.js#L15-L17

This is a clever way to reuse the same public handler for all your static pages. However it's not really what redirects are for.

Remember a redirect ends up requiring twice as many requests: the browser asks for `/`, gets a `302` response, _then_ grabs the `location` header and makes a whole 'nother request for that.

It's also a little weird for a user to go to `/` and end up at `/public/index.html`.

You could get the same reusability from your public handler by abstracting it into a function you can pass a file path to.

**Comment by Joepock123** [comment]
Not quite sure what you mean in the last sentence so will follow up if that's ok! @oliverjam

**Comment by oliverjam** [comment]
Yeah sorry I wasn't that clear. So you could for example add a third argument to your public handler to specify an exact file path, then use that instead of `request.url` if it's passed in:

```js
function home(request, response) {
  public(request, response, "public/index.html");
}

function public(request, response, filePath) {
  const url = filePath || request.url;
  // ... the same readFile stuff as before
}
```

So you can keep using the public handler for general static files, but also use it to serve a specific file when you need to.

**Comment by Joepock123** [comment]
That's a really cool idea. Thank you!
